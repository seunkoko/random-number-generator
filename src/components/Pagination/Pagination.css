.pagination__container {
  margin-bottom: 2rem;
  font-size: 0.6rem;
}

.pagination>.active>a {
  background-color: #222222;
  color: #F2F2F2 !important;
  border-color: #222222;
}

.pagination>.active>a:focus {
  background-color: #222222;
  color: #F2F2F2 !important;
  border-color: #222222;
}

.pagination>.active>a:hover {
  background-color: #222222;
  color: #F2F2F2 !important;
  border-color: #222222;
}

.pagination>.active>span {
  background-color: #222222;
  color: #F2F2F2 !important;
  border-color: #222222;
}

.pagination>.active>span:focus {
  background-color: #222222;
  color: #F2F2F2 !important;
  border-color: #222222;
}

.pagination>.active>span:hover {
  color: #222222 !important;
  border-color: #222222;
}

.pagination>li>a {
  color: #222222 !important;
}

@media screen and (max-width: 480px) {
  .pagination__container {
    width: 90%;
  }
}
