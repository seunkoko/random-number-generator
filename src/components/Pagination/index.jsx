// react library
import React from 'react';

// third-party libraries
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';

// styles
import './Pagination.css';

// utils
import baseRouterProps from '../../utils/baseRouterProps';

const paginateNav = (totalPages, handlePageClick) => (
  <ReactPaginate
    id="reactPaginateId"
    previousLabel="<"
    nextLabel=">"
    breakLabel={<a href="">...</a>} // eslint-disable-line jsx-a11y/anchor-is-valid
    breakClassName="break-me"
    pageCount={totalPages} // eslint-disable-line no-unneeded-ternary
    marginPagesDisplayed={3}
    pageRangeDisplayed={totalPages > 9 ? 10 : totalPages}
    onPageChange={handlePageClick}
    containerClassName="pagination justify-content-center"
    pageClassName="page-item"
    pageLinkClassName="page-link"
    nextClassName="page-item next-button"
    previousClassName="page-item"
    previousLinkClassName="page-link"
    nextLinkClassName="page-link"
    disabledClassName="disabled"
    activeClassName="active"
  />
);

/**
 * Renders Pagination component
 *
 * @param {object} props - component's own props
 * @returns {JSX} JSX
 */
const Pagination = (props) => {
  const {
    totalPages,
    handlePageClick,
  } = props;

  return (
    <nav className="container pagination__container">
      {totalPages > 1 ? paginateNav(totalPages, handlePageClick) : null}
    </nav>
  );
};

Pagination.propTypes = {
  ...baseRouterProps,
  totalPages: PropTypes.number.isRequired,
  handlePageClick: PropTypes.func.isRequired,
};

export default Pagination;
