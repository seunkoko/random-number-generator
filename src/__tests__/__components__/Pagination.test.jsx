// react library
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// components
import Pagination from '../../components/Pagination';

describe('Pagination', () => {
  let props = {
    totalPages: 0,
    handlePageClick: jest.fn(),
  };

  it('renders properly when props are supplied', () => {
    const wrapper = shallow(<Pagination {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when totalPages is greater than 0', () => {
    props = {
      ...props,
      totalPages: 100,
    };
    const wrapper = shallow(<Pagination {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when totalPages is less than 10', () => {
    props = {
      ...props,
      totalPages: 2,
    };
    const wrapper = shallow(<Pagination {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
