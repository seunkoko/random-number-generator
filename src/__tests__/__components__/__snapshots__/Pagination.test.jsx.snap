// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders properly when props are supplied 1`] = `
<nav
  className="container pagination__container"
/>
`;

exports[`Pagination renders properly when totalPages is greater than 0 1`] = `
<nav
  className="container pagination__container"
>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="break-me"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    id="reactPaginateId"
    marginPagesDisplayed={3}
    nextClassName="page-item next-button"
    nextLabel=">"
    nextLinkClassName="page-link"
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={100}
    pageLinkClassName="page-link"
    pageRangeDisplayed={10}
    previousClassName="page-item"
    previousLabel="<"
    previousLinkClassName="page-link"
  />
</nav>
`;

exports[`Pagination renders properly when totalPages is less than 10 1`] = `
<nav
  className="container pagination__container"
>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="break-me"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    id="reactPaginateId"
    marginPagesDisplayed={3}
    nextClassName="page-item next-button"
    nextLabel=">"
    nextLinkClassName="page-link"
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={2}
    pageLinkClassName="page-link"
    pageRangeDisplayed={2}
    previousClassName="page-item"
    previousLabel="<"
    previousLinkClassName="page-link"
  />
</nav>
`;
